src/Ext/BuildingType/Hooks.cpp

```diff
@@ -59,4 +59,192 @@ DEFINE_HOOK(0x458623, BuildingClass_KillOccupiers_Replace_MuzzleFix, 0x7)
 	}
 
 	return 0;
+}
+
+DEFINE_HOOK(0x6FE3F1, TechnoClass_Fireat_OccupyDamageBonus, 0xB)
+{
+	enum {
+		ApplyDamageBonus = 0x6FE405,
+		Nothing = 0x0
+	};
+
+	GET(TechnoClass* const, pThis, ESI);
+
+	if (auto const Building = specific_cast<BuildingClass*>(pThis))
+	{
+		if (auto const TypeExt = BuildingTypeExt::ExtMap.Find(Building->Type))
+		{
+			GET_STACK(int, nDamage, 0x2C);
+			R->EAX(Game::F2I(nDamage * TypeExt->BuildingOccupyDamageMult.Get(RulesClass::Instance->OccupyDamageMultiplier)));
+			return ApplyDamageBonus;
+		}
+	}
+
+	return Nothing;
+}
+
+DEFINE_HOOK(0x6FE421, TechnoClass_Fireat_BunkerDamageBonus, 0xB)
+{
+	enum {
+		ApplyDamageBonus = 0x6FE435,
+		Nothing = 0x0
+	};
+
+	GET(TechnoClass* const, pThis, ESI);
+
+	if (auto const Building = specific_cast<BuildingClass*>(pThis->BunkerLinkedItem))
+	{
+		if (auto const TypeExt = BuildingTypeExt::ExtMap.Find(Building->Type))
+		{
+			GET_STACK(int, nDamage, 0x2C);
+			R->EAX(Game::F2I(nDamage * TypeExt->BuildingBunkerDamageMult.Get(RulesClass::Instance->OccupyDamageMultiplier)));
+			return ApplyDamageBonus;
+		}
+	}
+
+	return Nothing;
+}
+
+DEFINE_HOOK(0x6FD183, TechnoClass_RearmDelay_BuildingOccupyROFMult, 0xC)
+{
+	enum {
+		ApplyRofMod = 0x6FD1AB,
+		SkipRofMod = 0x6FD1B1,
+		Nothing = 0x0
+	};
+
+	GET(TechnoClass*, pThis, ESI);
+
+	if (auto const Building = specific_cast<BuildingClass*>(pThis))
+	{
+		if (auto const TypeExt = BuildingTypeExt::ExtMap.Find(Building->Type))
+		{
+			auto const nMult = TypeExt->BuildingOccupyROFMult.Get(RulesClass::Instance->OccupyROFMultiplier);
+			if (nMult > 0.0f)
+			{
+				GET_STACK(int, nROF, STACK_OFFS(0x10, -0x4));
+				R->EAX(Game::F2I(((double)nROF) / nMult));
+				return ApplyRofMod;
+			}
+			return SkipRofMod;
+		}
+	}
+
+	return Nothing;
+}
+
+DEFINE_HOOK(0x6FD1C7, TechnoClass_RearmDelay_BuildingBunkerROFMult, 0xC)
+{
+	enum {
+		ApplyRofMod = 0x6FD1EF,
+		SkipRofMod = 0x6FD1F1,
+		Nothing = 0x0
+	};
+
+	GET(TechnoClass*, pThis, ESI);
+
+	if (auto const Building = specific_cast<BuildingClass*>(pThis->BunkerLinkedItem))
+	{
+		if (auto const TypeExt = BuildingTypeExt::ExtMap.Find(Building->Type))
+		{
+			auto const nMult = TypeExt->BuildingBunkerROFMult.Get(RulesClass::Instance->BunkerROFMultiplier);
+			if (nMult > 0.0f)
+			{
+				GET_STACK(int, nROF, STACK_OFFS(0x10, -0x4));
+				R->EAX(Game::F2I(((double)nROF) / nMult));
+				return ApplyRofMod;
+			}
+			return SkipRofMod;
+		}
+	}
+
+	return Nothing;
+}
+
+DEFINE_HOOK_AGAIN(0x45933D, BuildingClass_BunkerSound, 0x5)
+DEFINE_HOOK_AGAIN(0x4595D9, BuildingClass_BunkerSound, 0x5)
+DEFINE_HOOK(0x459494, BuildingClass_BunkerSound, 0x5)
+{
+	enum {
+		BunkerWallUpSound = 0x45933D,
+		BunkerWallUpSound_Handled_ret = 0x459374,
+
+		BunkerWallDownSound_01 = 0x4595D9,
+		BunkerWallDownSound_01_Handled_ret = 0x459612,
+
+		BunkerWallDownSound_02 = 0x459494,
+		BunkerWallDownSound_02_Handled_ret = 0x4594CD
+
+	};

```

Comment:
```suggestion
	enum
	{
		BunkerWallUpSound = 0x45933D,
		BunkerWallUpSound_Handled_ret = 0x459374,

		BunkerWallDownSound_01 = 0x4595D9,
		BunkerWallDownSound_01_Handled_ret = 0x459612,

		BunkerWallDownSound_02 = 0x459494,
		BunkerWallDownSound_02_Handled_ret = 0x4594CD
	};
```

Also if you're defining hook addresses in the enum - maybe it could be good to have the enum defined outside of the hook and reused in hook definition macro? assuming that is possible at all